Controllers/ProductsController.cs

```diff
@@ -0,0 +1,217 @@
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using StoresManagement.Data;
+using StoresManagement.Models;
+using StoresManagement.ViewModels;
+
+namespace StoresManagement.Controllers
+{
+    public class ProductsController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IMapper _mapper;
+
+        public ProductsController(ApplicationDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        // GET: Products
+        public async Task<IActionResult> Index()
+        {
+            var Products = await _context.Products
+                .Include(b => b.Branch)
+                .Include(b => b.Branch.Entity)
+                .ToListAsync();
+
+            var branchesVM = new List<ProductFormViewModel>();
+
+            foreach (var product in Products)
+            {
+                branchesVM.Add(_mapper.Map<ProductFormViewModel>(product));
+            }
+
+            return View(branchesVM);
+        }
+
+        // GET: Products by product Id
+        public async Task<IActionResult> ListProducts(int? id)
+        {
+            var Products = await _context.Products
+                .Include(b => b.Branch)
+                .Include(b => b.Branch.Entity)
+                .Where(m => m.BranchId == id)
+                .ToListAsync();
+
+            var branchesVM = new List<ProductFormViewModel>();
+
+            foreach (var product in Products)
+            {
+                branchesVM.Add(_mapper.Map<ProductFormViewModel>(product));
+            }
+
+            return View(branchesVM);
+        }
+
+        // GET: Products/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var product = await _context.Products
+                .Include(b => b.Branch)
+                .Include(b => b.Branch.Entity)
+                .FirstOrDefaultAsync(m => m.Id == id);
+
+            if (product == null)
+            {
+                return NotFound();
+            }
+
+            return View(_mapper.Map<ProductFormViewModel>(product));
+        }
+
+        // GET: Products/Create
+        public IActionResult Create()
+        {
+            var productVM = new ProductFormViewModel
+            {
+                Branches = _context.Branches.ToList()
+            };
+
+            return View(productVM);
+        }
+
+        // POST: Products/Create
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create(ProductFormViewModel productVM)
+        {
+            if (ModelState.IsValid)
+            {
+                var branch = await _context.Branches
+            .FirstOrDefaultAsync(m => m.Id == productVM.BranchId);
```

Comment:
formatting. 
I guess you need to take care of this indentation on your own. For other cases use sequence
`ctrl+k, ctrl+d` in VS will autoformat code